### PR TITLE
Refactor search pipeline

### PR DESCRIPTION
**Pull request by Weves**

### CODE REVIEW

File: backend/danswer/llm/answering/prompts/citations_prompt.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
+from collections.abc import Callable
```
Comment (yuhongsun96) [inline backend/danswer/llm/answering/prompts/citations_prompt.py:1]: calling these citation_prompt/quote_prompt is a bit misleading as the are not prompt files

File: backend/danswer/llm/answering/answer.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+class Answer:
+    def __init__(
+        self,
+        question: str,
+        docs: list[LlmDoc],
+        answer_style_config: AnswerStyleConfig,
+        prompt: Prompt,
+        persona: Persona,
+        # must be the same lenght as `docs`. If None, all docs are considered "relevant"
```
Comment (yuhongsun96) [inline backend/danswer/llm/answering/answer.py]: REEE lenght, get a spellchecker plugin!

File: backend/danswer/llm/answering/prompts/citations_prompt.py (hunk 1 above)
Comment (Weves) [reply backend/danswer/llm/answering/prompts/citations_prompt.py:1]: we have utils.py under `prompts` already—I see this just as a designation that what is in this folder has to do with prompts / creating prompts / managing prompts

File: backend/danswer/llm/answering/answer.py (hunk 2 above)
Comment (Weves) [reply backend/danswer/llm/answering/answer.py]: insane eye :O

**Pull request merged**